oremda/pipeline/runner/context.py

```diff
@@ -0,0 +1,72 @@
+from contextlib import contextmanager
+
+from typing import Dict, Set
+
+from pydantic import BaseModel, Field
+
+from oremda.typing import IdType
+from oremda.clients import Client as ContainerClientFactory
+from oremda.clients.base import ClientBase as ContainerClient
+from oremda.plasma_client import PlasmaClient
+from oremda.registry import Registry
+from oremda.constants import DEFAULT_OREMDA_VAR_DIR
+from oremda.utils.plasma import start_plasma_store
+from oremda.typing import ContainerType
+from oremda.pipeline.runner.config import settings
+
+from oremda.models import (
+    SessionModel,
+    WebsocketModel,
+)
+
+from oremda.pipeline.models import (
+    PipelineModel,
+)
+
+
+class GlobalContext(BaseModel):
+    plasma_client: PlasmaClient = Field(...)
+    container_client: ContainerClient = Field(...)
+    registry: Registry = Field(...)
+    sessions: Dict[IdType, SessionModel] = {}
+    pipelines: Dict[IdType, PipelineModel] = {}
+    websockets: Dict[IdType, WebsocketModel] = {}
+    session_pipelines: Dict[IdType, Set[IdType]] = {}
+    session_websockets: Dict[IdType, Set[IdType]] = {}
```

Comment:
Just an idea here (that we don't necessarily need to do). What if we had a model like this:
```python
class SessionWebModel(BaseModel):
    session: SessionModel
    pipelines: Set[IdType] = set()
    websockets: Set[IdType] = set()
```

This way we wouldn't need `session_pipelines` or `session_websockets`.

Reply:
This is copied code, but I can make that improvement.

Reply:
Done

Reply:
This is a good improvement so that we do not have to worry about keeping them synchronized.